xray/src/bin/inpaint_xray_quadtree.rs

```diff
@@ -0,0 +1,158 @@
+            if let Ok(neighbor_meta) = Meta::from_disk(get_meta_pb_path(
+                input_directory,
+                NodeId::from(spatial_root_neighbor_id),
+            )) {
+                for neighbor_node_id in neighbor_meta
+                    .nodes
+                    .iter()
+                    .filter(|id| id.level() == neighbor_meta.deepest_level)
+                {
+                    if let Some(spatial_id) =
+                        SpatialNodeId::from(*neighbor_node_id).neighbor(direction.opposite())
+                    {
+                        if leaf_node_ids.contains(&NodeId::from(spatial_id)) {
+                            neighboring_leaf_node_ids.insert(*neighbor_node_id);
+                        }
+                    }
+                }
+            }
+        }
+    }
+    neighboring_leaf_node_ids
+}
+
+fn copy_nodes(
+    node_ids: &FnvHashSet<NodeId>,
+    input_directory: &Path,
+    output_directory: &Path,
+) -> io::Result<()> {
+    for node_id in node_ids.iter() {
+        fs::copy(
+            get_image_path(input_directory, *node_id),
+            get_image_path(output_directory, *node_id),
+        )?;
+    }
+    Ok(())
+}
+
+fn copy_meta_pb(
+    root_node_id: NodeId,
+    input_directory: &Path,
+    output_directory: &Path,
+) -> io::Result<u64> {
+    fs::copy(
+        get_meta_pb_path(&input_directory, root_node_id),
+        get_meta_pb_path(&output_directory, root_node_id),
+    )
+}
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let args = CommandlineArguments::from_args();
+    let input_directory = args.input_directory.canonicalize()?;
+    if !args.output_directory.exists() {
+        fs::create_dir_all(&args.output_directory)?;
+    }
+    let output_directory = args.output_directory.canonicalize()?;
+    let tile_background_color = args.tile_background_color.to_color();
+    let root_node_id = args.root_node_id;
+    let meta = Meta::from_disk(get_meta_pb_path(&input_directory, root_node_id))?;
+
+    let leaf_node_ids: FnvHashSet<NodeId> = meta
+        .nodes
+        .iter()
+        .copied()
+        .filter(|id| id.level() == meta.deepest_level)
+        .collect();
+
+    let adjacent_leaf_node_ids =
+        get_adjacent_leaf_node_ids(&leaf_node_ids, &input_directory, root_node_id);
+    if root_node_id.level() != 0 && adjacent_leaf_node_ids.is_empty() {
+        println!(
+            "No adjacent leaf nodes found in neighboring quadtrees. \
+             Did you forget to copy them into {:?}?",
+            &input_directory
+        )
+    }
+
+    if input_directory != output_directory {
+        copy_meta_pb(root_node_id, &input_directory, &output_directory)?;
+        copy_nodes(&leaf_node_ids, &input_directory, &output_directory)?;
+        copy_nodes(&adjacent_leaf_node_ids, &input_directory, &output_directory)?;
+    }
+
+    perform_inpainting(&output_directory, args.inpaint_distance_px, &leaf_node_ids)?;
+
```

Comment:
nit: remove this and the next empty line?

Reply:
Done